vignettes/dev04_RECIPES.Rmd

```diff
@@ -0,0 +1,61 @@
+```
+
+The naming of the table is important! When pulling this data into the application S3 objects are heavily utilized. In this example, the table object will be given the class `stan_3`. The application will use this information to correctly process the data used in the creation of the table.
+
+The table object can contain a number of sub-objects. In the example, there are three such objects with the following key names: `title`, `group_by`, and `blocks`. The `title` object contains the text to be used as the title in the table generation. The `group_by` object contains the name of the field that the table generator should use to group by. The `blocks` object is more complicated.
+
+### The `blocks` object
+
+The `blocks` object is an array of `block` objects. The setup is very similar to blocks used in the Table Generator. The `blocks` object and the blocks from the Table Generator are two sides of the same coin, hence the naming convention used. Each `block` is a named list that can contain a number of different elements. Some of the most important elements are included in the example: `data`, `variable`, and `statistic`.
+
+The `data` element denotes which data set provided is being used to create the block. The `variable` element denotes the field or parameter from the data set used for the block's statistical computation. The `statistic` element denotes which statistic is being calculated for the field/parameter.
+
+A few other elements can also be present within a `block` object. The `recipes.json` file contains a couple more: `stat_arg` and `stat_options_fn`.  Currently no other elements have usage in the application, but active development will be changing this soon.
+
+## The parsing of `recipes.json`
+
+### The table object
+
+As stated previously, the key/name of the table object in `recipes.json` is assigned as a class when pulling the table object into R. This class is used to determine how the pre-filtering should be completed in the functions `filter_adsl()` and `filter_adae()`. New outputs can be created for a hypothetical table with key `tbl_01` by defining `filter_adsl.tbl_01()` and `filter_adae.tbl_01()`.
+
+The key/name of the table object will also be used to define the option in the recipe select input in the application.
+
+### The `blocks` object
+
+There is one additional sub-object not yet mentioned, the `recipe_inclusion` object, which can assign a class to the `blocks` object. The standard inclusion criteria is for all data sets in a `blocks` object to be present in the list of data sets uploaded and for all variables listed in the `block` objects to be present in the data set identified with them. Assigning a value to `recipe_inclusion` can affect the rules for inclusion in the recipes select input HTML element. If a user had a special set of criteria, they could assign `crit_01` to `recipe_inclusion` and assign the function `recipe_inclusion.crit_01()` to determine the criteria.
```

Comment:
I think this paragraph deserves an example. I saw below that it's associated with table 41, but I'm still not sure I understand what it does. What's an example for your example on `crit_01`? lol